### DISCUSSION TITLE
`Segmentation fault` immediately when lifetime sets up `asyncio.eager_task_factory`

### DISCUSSION THREAD

**Discussion by Zac-HD** [fastapi/fastapi#11868] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import asyncio
from contextlib import asynccontextmanager
from fastapi import FastAPI
from fastapi.testclient import TestClient

@asynccontextmanager
async def lifespan(_: FastAPI):
    loop = asyncio.get_event_loop()
    loop.set_task_factory(asyncio.eager_task_factory)
    yield

app = FastAPI(title="test", lifespan=lifespan, debug=True)

@app.get("/tst")
async def endpoint() -> dict[str, str]:
    return {"message": "Hello World"}

def test():
    with TestClient(app=app) as client:
        client.get("/tst")

if __name__ == "__main__":
    test()
```

### Description

Running this code - with Python 3.12, which added `asyncio.eager_task_factory` - immediately crashes with `Segmentation fault`.

### Operating System

Linux

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

3.12.4

### Additional Context

Initially reported in https://github.com/pytest-dev/pytest/issues/12570 with a more complicated reproducer; I hope this is clear enough to be helpful!

**Comment by svlandeg** [comment]
PR with unit test (though not yet a fix): https://github.com/fastapi/fastapi/pull/12079

**Comment by tiangolo** [comment, marked as answer]
Thanks for taking the time to make the minimal example and report it here @Zac-HD! :coffee: 

I was able to reproduce it without FastAPI, with pure Starlette, so it seems the problem is upstream. I suspect AnyIO or asyncio. I reported it here: https://github.com/encode/starlette/discussions/2890
